mlflow/xgboost.py

```diff
@@ -130,7 +138,22 @@ def save_model(
                           model. The given example will be converted to a Pandas DataFrame and then
                           serialized to json using the Pandas split-oriented format. Bytes are
                           base64-encoded.
+    :param pip_requirements: Overwrites pip requirements of the conda environment this model should
```

Comment:
`pip_requirements` should control which requirements are logged to `requirements.txt`. This is more important for users than the conda environment behavior. We should emphasize `requirements.txt` and say that the conda environment behavior is there for backwards compatibility.

Reply:
I tried this out manually with the following code: 

```
import mlflow

import xgboost as xgb
from sklearn import datasets
from sklearn.model_selection import train_test_split

iris = datasets.load_iris()
X = iris.data
y = iris.target
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

dtrain = xgb.DMatrix(X_train, label=y_train)
dtest = xgb.DMatrix(X_test, label=y_test)


params = {
    "objective": "multi:softprob",
    "num_class": 3,
    "eval_metric": "mlogloss",
    "seed": 42,
}
model = xgb.train(params, dtrain, evals=[(dtrain, "train")])

mlflow.xgboost.log_model(
    model,
    additional_pip_requirements=[
        "foobaz==barbiz",
    ],
    artifact_path="foobiz",
)
```

The resulting requirements.txt doesn't contain my test requirement:

```
mlflow
xgboost==1.5.0-dev
```

However, the conda environment is correct:

```
channels:
- conda-forge
dependencies:
- python=3.8.5
- pip
- pip:
  - mlflow
  - xgboost==1.5.0-dev
  - foobaz==barbiz
name: mlflow-env
```

Reply:
Strange, I'll take a look.